src/C++/FieldConvertors.h

```diff
@@ -342,18 +339,11 @@ static bool convert( const std::string& value, double& result )
   }
 
   if( *i || !haveDigit ) return false;
-    
+
   int processed_chars;
   const int total_length = value.length();
-  const double val = fast_strtod( value.c_str(), total_length, &processed_chars);
-
-  if ( processed_chars != total_length ||
-     val != val /*test for quite NaN*/ )
```

Comment:
so, we don't need to check for NaN?

Reply:
As from the above comments (hard to tell because github cuts off alot the changed code on comments)

This check is redundant and unreachable due to the section of code that proceeds this area as shown

``
const char * i = value.c_str();

// Catch null strings
if( !*i ) return false;
// Eat leading '-' and recheck for null string
if( i == '-' && !++i ) return false;

bool haveDigit = false;

if( IS_DIGIT(i) )
{
haveDigit = true;
while( IS_DIGIT (++i) );
}

if( i == '.' && IS_DIGIT(++i) )
{
haveDigit = true;
while( IS_DIGIT (*++i) );
}

if( *i || !haveDigit ) return false;
``

This chunk does all the required checks to ensure the inputted value can be converted. Rechecking an output from a helper function is not necessary due to the fact that this helper function cannot receive a value that cannot be converted.

Reply:
> so, we don't need to check for NaN?

@OleksiyLevit, this stood out to me too, but `gcov` says this check is unreachable. If that doesn't hold for other compilers/platforms, then we should reconsider.